### PR TITLE
Selinux loading fix and warning cleanups

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by evverx** [comment]
@keszybz , thanks for the `test-selinux`.
I've found a problem:

```
$ sudo ./test-selinux
...
mac_selinux_create_label_from_exe → 0 (No such file or directory), "unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023"
...

$ sudo strace ./test-selinux
...
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
open("/sys/fs/selinux/create", O_RDWR)  = 3
write(3, "unconfined_u:unconfined_r:unconf"..., 88) = 88
read(3, "unconfined_u:unconfined_r:unconf"..., 4095) = 54
close(3)                                = 0
socket(PF_LOCAL, SOCK_STREAM|SOCK_CLOEXEC, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
writev(2, [{"mac_selinux_create_label_from_ex"..., 124}, {"\n", 1}], 2mac_selinux_create_label_from_exe → 0 (No such file or directory), "unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023"
...

$ sudo systemctl start mcstransd.service
$ sudo ./test-selinux
...
mac_selinux_create_label_from_exe → 0 (Success), "unconfined_u:unconfined_r:unconfined_t:SystemLow-SystemHigh"
...
```

So, we use non-raw functions sometimes. It's not good: https://github.com/systemd/systemd/pull/1107

File: src/basic/selinux-util.c [hunk 1]
```diff
@@ -225,7 +217,7 @@ int mac_selinux_get_create_label_from_exe(const char *exe, char **label) {
                 return -errno;
 
         sclass = string_to_security_class("process");
-        r = security_compute_create(mycon, fcon, sclass, (security_context_t *) label);
+        r = security_compute_create_raw(mycon, fcon, sclass, (security_context_t *) label);
```
Comment (evverx) [inline src/basic/selinux-util.c:220]: @keszybz , thanks!
I'm running `exec strace -D -ff -o strace-of-systemd /usr/lib/systemd/systemd` as `init`:

```
bash-4.3# grep -F '.setrans-unix' /strace-of-systemd.1
connect(19, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(22, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(22, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(23, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(23, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(25, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(26, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(28, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(29, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(39, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(39, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(34, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(34, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(35, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(35, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(15, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
connect(15, {sa_family=AF_LOCAL, sun_path="/var/run/setrans/.setrans-unix"}, 110) = -1 ENOENT (No such file or directory)
```

So, it's not so easy:)

Comment (keszybz) [inline src/basic/selinux-util.c:220]: I'm not seeing that here (running with 6f81773de7fe5ad4).

Comment (evverx) [inline src/basic/selinux-util.c:220]: Hm, it's strange.
For example: `systemd` [calls](https://github.com/systemd/systemd/blob/c3dacc8bbf2dc2f5d498072418289c3ba79160ac/src/core/main.c#L1359) `mac_selinux_setup`. `mac_selinux_setup` [calls](https://github.com/systemd/systemd/blob/c3dacc8bbf2dc2f5d498072418289c3ba79160ac/src/core/selinux-setup.c#L91) `setcon`. `setcon` calls `selinux_raw_to_trans_context`.

Comment (evverx) [inline src/basic/selinux-util.c:220]: I've prepared a PR: https://github.com/systemd/systemd/pull/2797

**Pull request merged**